vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditReferenceController.java

```diff
@@ -0,0 +1,88 @@
+    
+    @RequestMapping(value = "/staff/reference/{referenceId}/edit", method = RequestMethod.GET)
+    public String show(Model model, @PathVariable("referenceId") String referenceId, RedirectAttributes attributes) {
+        IReference reference = referenceManager.getReferenceById(referenceId);
+        model.addAttribute("referenceData", reference);
+        model.addAttribute("referenceTypes", ReferenceType.values());
+        return "staff/references/edit";
+    }
+    
+    /**
+     * This method allows updating References when a user is on the reference page accessed from 'References' option from side navbar.
+     */
+    @RequestMapping(value = "/staff/reference/{referenceId}/edit", method = RequestMethod.POST)
+    public String save(@ModelAttribute Reference refData, @PathVariable("referenceId") String referenceId, 
+            RedirectAttributes attributes) {
+        IReference reference = referenceManager.getReferenceById(referenceId);
+        if(reference!=null) {
+            refData.setId(reference.getId());
+            refData.setCreatedBy(reference.getCreatedBy());
+            refData.setCreationDate(reference.getCreationDate());
+            referenceManager.updateReference(refData);
+            attributes.addFlashAttribute("alertType", "success");
+            attributes.addFlashAttribute("message", "Reference successfully updated!");
+            attributes.addFlashAttribute("showAlert", "true");
+            return "redirect:/staff/display/reference/{referenceId}";
+        }
+        return "redirect:/404";
+    }
+    
+    /**
+     * This method allows updating References when a user is on the slide page.
+     */
+    @RequestMapping(value = "/staff/module/{moduleId}/slide/{id}/biblio/{biblioId}/reference/{refId}/edit", method = RequestMethod.POST)
+    public ResponseEntity<ReferenceBlock> editReference(@PathVariable("id") String slideId,
+            @PathVariable("moduleId") String moduleId, @PathVariable("biblioId") String biblioId, 
+            @PathVariable("refId") String refId, @RequestBody Reference ref, RedirectAttributes attributes) throws IOException {
+        IReference reference = referenceManager.getReference(refId);
+        if(reference!=null) {
+            ref.setId(reference.getId());
+            ref.setCreatedBy(reference.getCreatedBy());
+            ref.setCreationDate(reference.getCreationDate());
+            referenceManager.updateReference(ref);
+            String refDisplayText = referenceDisplayProvider.getReferenceDisplayText((Reference)ref);
+            ReferenceBlock refBlock = new ReferenceBlock(ref, refDisplayText);
+            attributes.addFlashAttribute("alertType", "success");
+            attributes.addFlashAttribute("message", "Reference successfully updated!");
+            attributes.addFlashAttribute("showAlert", "true");
+            return new ResponseEntity<ReferenceBlock>(refBlock, HttpStatus.OK);
```

Comment:
since you're returning a responseentity here, I don't think the flash attributes are doing anything useful.